src/main/java/sparksoniq/jsoniq/item/Item.java

```diff
@@ -87,25 +87,23 @@ public static Type getNumericResultType(Item left, Item right) {
     }
 
     //returns an effective boolean value of any item type
-    public static boolean getEffectiveBooleanValue(Item item) {
-        if (item == null)
+    public boolean getEffectiveBooleanValue() {
```

Comment:
Note if you want to do this even better: this method could actually be made abstract and overriden for each subclass.